features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/startup/RumAppStartupDetectorImplTest.kt

```diff
@@ -0,0 +1,436 @@
+        detector.onActivityStopped(activity)
+        detector.onActivityDestroyed(activity)
+
+        currentTime += 30.seconds
+
+        val activity2 = mock<Activity>()
+
+        triggerBeforeCreated(
+            forge = forge,
+            detector = detector,
+            activity = activity2,
+            hasSavedInstanceStateBundle = hasSavedInstanceStateBundle2
+        )
+
+        // Then
+        listener.verifyScenarioDetected(
+            RumStartupScenario.WarmAfterActivityDestroyed(
+                initialTimeNanos = currentTime.inWholeNanoseconds,
+                hasSavedInstanceStateBundle = hasSavedInstanceStateBundle2,
+                activity = activity2
+            )
+        )
+
+        verifyNoMoreInteractions(listener)
+    }
+
+    @Test
+    fun `M detect WarmAfterActivityDestroyed W RumAppStartupDetector {2 activities created, destroyed and 3rd created}`(
+        forge: Forge,
+        @BoolForgery hasSavedInstanceStateBundle: Boolean,
+        @BoolForgery hasSavedInstanceStateBundle2: Boolean,
+        @BoolForgery hasSavedInstanceStateBundle3: Boolean
+    ) {
+        // Given
+        val detector = createDetector(
+            processImportance = IMPORTANCE_FOREGROUND
+        )
+
+        launchColdAndVerify(
+            forge = forge,
+            detector = detector,
+            hasSavedInstanceStateBundle = hasSavedInstanceStateBundle
+        )
+
+        // When
+        detector.onActivityStarted(activity)
+        detector.onActivityResumed(activity)
+        detector.onActivityPaused(activity)
+        detector.onActivityStopped(activity)
+
+        currentTime += 30.seconds
+
+        val activity2 = mock<Activity>()
+
+        triggerBeforeCreated(
+            forge = forge,
+            detector = detector,
+            activity = activity2,
+            hasSavedInstanceStateBundle = hasSavedInstanceStateBundle2
+        )
+
+        verifyNoMoreInteractions(listener)
+
+        detector.onActivityStarted(activity2)
+        detector.onActivityResumed(activity2)
+        detector.onActivityPaused(activity2)
+        detector.onActivityStopped(activity2)
+        detector.onActivityDestroyed(activity2)
+
+        detector.onActivityDestroyed(activity)
+
+        currentTime += 30.seconds
+
+        val activity3 = mock<Activity>()
+
+        triggerBeforeCreated(
+            forge = forge,
+            detector = detector,
+            activity = activity3,
+            hasSavedInstanceStateBundle = hasSavedInstanceStateBundle3
+        )
+
+        // Then
+        listener.verifyScenarioDetected(
+            RumStartupScenario.WarmAfterActivityDestroyed(
+                initialTimeNanos = currentTime.inWholeNanoseconds,
+                hasSavedInstanceStateBundle = hasSavedInstanceStateBundle3,
+                activity = activity3
+            )
+        )
+
+        verifyNoMoreInteractions(listener)
+    }
+
+    private fun launchColdAndVerify(
```

Comment:
I would propose to not use a method which has "Given + When + Then" inside, and then after its call there is another "When + Then", because it is hard to read the overall test and its flow.

Maybe you can just copy-paste its contents directly where it is used? Yes, it will be some repetition, but it will be easier to read.